tests/integration/conftest.py

```diff
@@ -1078,3 +994,90 @@ def keep_centos_pointed_to_vault(shell):
         sed_repos_to_vault = r'sed -i -e "s|^\(mirrorlist=.*\)|#\1|" -e "s|^#baseurl=http://mirror\(.*\)|baseurl=http://vault\1|" /etc/yum.repos.d/CentOS-*'
         print("TESTS >>> Resetting the repos to vault")
         shell(sed_repos_to_vault, silent=True)
+
+
+class Satellite:
+    def __init__(self, key):
+        self.shell = live_shell()
+        # Key on which upon the command is selected
+        self.key = key if key else SYSTEM_RELEASE_ENV
+        # File containing registration commands
+        self._sat_reg_commands = dotenv_values("/var/tmp/.env_sat_reg")
+        self._sat_script_location = "/var/tmp/register_to_satellite.sh"
+        self.subman = SubscriptionManager()
+
+    def get_satellite_curl_command(self):
+        """
+        Get the Satellite registration command for the respective system.
+        """
+        if not self._sat_reg_commands:
+            pytest.fail(
+                f"The {self._sat_reg_file} either not found or empty.\
+                It is required for the satellite conversion to work."
+            )
+
+        return self._sat_reg_commands.get(self.key)
+
+    def _curl_the_satellite_script(self, curl_command):
+        assert (
+            self.shell(f"{curl_command} -o {self._sat_script_location}", silent=True).returncode == 0
+        ), "Failed to curl the satellite script to the machine."
+
+    def _run_satellite_reg_script(self):
+        assert (
+            self.shell(f"chmod +x {self._sat_script_location} && /bin/bash {self._sat_script_location}").returncode == 0
+        ), "Falied to run the satellite registration script."
+
+    def register(self):
+        curl_command = self.get_satellite_curl_command()
+
+        if "oracle" in SYSTEM_RELEASE_ENV:
+            self.subman.add_keys_and_certificates()
+            self.subman.add_client_tools_repo()
+
+        # Make sure it returned some value, otherwise it will fail.
+        assert curl_command, "The registration command is empty."
+
+        # Curl the Satellite registration script silently
+        self._curl_the_satellite_script(curl_command)
+
+        # Make the script executable and run the registration
+        self._run_satellite_reg_script()
+
+        if "oracle" in SYSTEM_RELEASE_ENV:
+            self.subman.remove_client_tools_repo()
```

Comment:
A comment would be nice again.
(I know the previous fixture was lacking the comments in some places, but that's the point of the refactor, right? I mean making it better :grimacing: )